### DISCUSSION TITLE
out-of-memory issue

### DISCUSSION THREAD

**Discussion by eric-yu-zhu** [CederGroupHub/chgnet#120] [Q&A] [answered]
My experience with chgnet is great. Thanks the team for developing and sharing it!
 
Recently, I got an out-of-memory issue while trying to relax a structure of 12,000 atoms. The experiment was done on an A6000 GPU having 48 GB memory. Below are a few questions:

Q1: What are the memory bottlenecks of chgnet?

Q2: Do you have a benchmark for memory usage of chgnet? My tests seem to show that 4000-atom-structure uses up 30 GB and 5000-atom-structure exceeds 48 GB. Make sense?

Q3: If the GPU memory issue cannot be resolved, should I try CPU version of chgnet? How many CPU cores are equivalent to the computing power of one decent GPU (e.g. V100) for running chgnet?

Any other suggestions?

**Comment by bowen-bd** [comment, marked as answer]
Q1,Q2 your test results are reasonable. I expect A100 to be able to simulate ~10K atoms. the exact number can vary based on the structure

Q3 I haven't done systematic tests on this, my experience is CPUs are significantly slower. For structure relaxations, CPUs might be OK, as they're not as costly as MDs.

**Comment by wugaxp** [comment]
I'm facing the same problem recently. In my case, my system contains around 5k atoms, and the gpu device is A100-40G, however out-of-memory error already appears.
Is it possible to run chgnet on multiple gpu cards? Perhaps that can solve the memory problem in some cases. At least in my case, we can access up to 4 A100-40G cards on each node.
